src/Data/List/Relation/Binary/Disjoint/DecSetoid.agda

```diff
@@ -0,0 +1,40 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Decidability of the disjoint relation over setoid equality.
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+open import Function.Base using (_$_)
+open import Data.Empty using (⊥-elim)
+open import Data.Product.Base using (_,_)
+open import Data.List.Base using ([]; _∷_; List)
+open import Data.List.Relation.Unary.Any using (here; there; map)
+open import Data.List.Relation.Unary.All using (all?; lookupₛ)
+open import Data.List.Relation.Unary.All.Properties using (¬All⇒Any¬)
+open import Relation.Binary.Bundles using (DecSetoid)
+open import Relation.Binary.Definitions using (Decidable)
+open import Relation.Nullary using (¬_; yes; no)
+open import Relation.Nullary using (¬?; _×-dec_)
+
+module Data.List.Relation.Binary.Disjoint.DecSetoid {c ℓ} (S : DecSetoid c ℓ) where
+
+open DecSetoid S
+open import Data.List.Relation.Binary.Equality.DecSetoid S
+open import Data.List.Relation.Binary.Disjoint.Setoid setoid public
+open import Data.List.Membership.DecSetoid S
+
+¬∉⇒∈ : ∀ {x xs} → ¬ (x ∉ xs) → x ∈ xs
```

Comment:
This lemma is `decidable-stable` for membership, being `Decidable` via `Any`... suggest refactoring along those lines?